subscription.go

```diff
@@ -103,6 +103,12 @@ func (r *marathonClient) registerSubscription() error {
 	case EventsTransportCallback:
 		return r.registerCallbackSubscription()
 	case EventsTransportSSE:
+		if timeout := r.client.config.HTTPClient.Timeout; timeout > 0 {
```

Comment:
by using [context](https://blog.golang.org/context) we will also give user explicit way to cancel any pending request - this will also give user option to set per request timeouts etc. But this change is epic in relative to this. This is just an example how this can be solved in much more elegant way then just fine tunning single shared client instance.